### ISSUE TITLE
Monitoring changes

### ISSUE THREAD

**Issue by Ivshti** [Softmotions/ejdb#41]
It would be great if there was a method for assigning a callback that should be called when changes occur in a collection. It would be even better if it accepted an argument that would allow you to filter the objects you wish to monitor (a filter query).

**Comment by almightycouch** [comment]
I wrote a small project using EJDB and i had to make some small changes to implement hooks with a filter query. I achieved that using the already built-in `tcxstrprintf()` LOG function.

Adding a LOG trace containing the modified BSON object for each updated/removed document:

    $UPDATE: 7b6173646173646173646173647d6b666a616b6364ba7b // BSON in HEX
    $UPDATE: 61736461f66b6a6173f66b666a616b6cf673646a617364 // BSON in HEX
    $DROPALL: 50585fbcb046b2709a534502 // OID as string

To evaluate which fields were modified, i used `ejdbqryexecute()` and `ejdbqresultbsondata()` to retrieve matching documents and combine them with the LOG of the executed query.

To allow an extra `query` argument to filter the object to monitor, i also implemented #164.


If you are interested in the patch needed in EJDB for this to work, have a look at my fork: https://github.com/Softmotions/ejdb/compare/master...almightycouch:meteorpp

For more informations, about the full implementation, have a look at the project: https://github.com/almightycouch/meteorpp
